crypto/evp/m_sigver.c

```diff
@@ -114,20 +170,39 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
      * Because we cleared out old ops, we shouldn't need to worry about
      * checking if signature is already there.
      */
-    signature = EVP_SIGNATURE_fetch(locpctx->libctx, supported_sig,
-                                    locpctx->propquery);
+    signature = evp_signature_fetch_from_prov((OSSL_PROVIDER *)keymgmt_prov,
```

Comment:
I am not sure this is right. The provider of the originally fetched keymgmt might not support a signature operation.

Reply:
I would rather keep a generic EVP_SIGNATURE_fetch here and try to export if needed.